exchanges/binance/binance_websocket.go

```diff
@@ -454,66 +455,158 @@ func (b *Binance) UpdateLocalBuffer(wsdp *WebsocketDepthStream) error {
 		return err
 	}
 
-	currencyPair, err := currency.NewPairFromFormattedPairs(wsdp.Pair,
+	currencyPair, err := currency.NewPairFromFormattedPairs(u.Pair,
 		enabledPairs,
 		format)
 	if err != nil {
 		return err
 	}
 
-	currentBook := b.Websocket.Orderbook.GetOrderbook(currencyPair, asset.Spot)
+	return b.CheckAndLoadBook(currencyPair, asset.Spot, u)
+}
+
+// CheckAndLoadBook buffers updates, spawns workers for http book request and
+// lets slip routine, validates buffer contents and returns required work when
+// needed
+func (b *Binance) CheckAndLoadBook(c currency.Pair, ass asset.Item, u *WebsocketDepthStream) error {
```

Comment:
None of these functions have tests. I understand that websocket tests can become more difficult but there are a lot of things going on that we shouldn't rely on eyeballing results since we are much more likely to miss things. This PR has also caused tests to fail. The best case would be to have a simulated test, but we also have the capability to do live test if needed